node-agent/internal/comms/comms_test.go

```diff
@@ -242,3 +242,30 @@ func TestProvisionReleaseServiceTokenAnonymous(t *testing.T) {
 	assert.Equal(t, relToken.AccessToken, token)
 	defer os.RemoveAll(testAuthClientCredsPath)
 }
+
+func TestProvisionReleaseServiceTokenError(t *testing.T) {
+	accessToken := "TEST-ACCESS-TOKEN"
+	authConf := getAuthConfig()
+
+	token := ""
+	ts := newTestIDPServer(t, http.MethodGet, token, "/token", http.StatusBadGateway)
+	defer ts.Close()
+
+	u, _ := url.Parse(ts.URL)
+
+	// Add mock server certificate to system cert pool
+	certPool, _ := x509.SystemCertPool()
+	certPool.AddCert(ts.Certificate())
+
+	ctx := context.Background()
+
+	// Now create release service token
+	relAuthCli, err := comms.GetAuthCli(u.Host, guid, certPool)
+	assert.Nil(t, err)
+
+	relToken, err := relAuthCli.ProvisionReleaseServiceToken(ctx, authConf, accessToken)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "502")
```

Comment:
Rather than having to hardcode the status code for a bad gateway, can we not use `http.StatusBadGateway` instead?